pinot-segment-local/src/main/java/org/apache/pinot/segment/local/aggregator/AvgMVValueAggregator.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.segment.local.aggregator;
+
+import javax.annotation.Nullable;
+import org.apache.pinot.segment.local.customobject.AvgPair;
+import org.apache.pinot.segment.local.utils.CustomSerDeUtils;
+import org.apache.pinot.segment.spi.AggregationFunctionType;
+import org.apache.pinot.spi.data.FieldSpec.DataType;
+
+
+/**
+ * Value aggregator for AVGMV aggregation function.
+ * This aggregator handles multi-value columns by computing the average across all values in all arrays.
+ */
+public class AvgMVValueAggregator implements ValueAggregator<Object, AvgPair> {
+  public static final DataType AGGREGATED_VALUE_TYPE = DataType.BYTES;
+
+  @Override
+  public AggregationFunctionType getAggregationType() {
+    return AggregationFunctionType.AVGMV;
+  }
+
+  @Override
+  public DataType getAggregatedValueType() {
+    return AGGREGATED_VALUE_TYPE;
+  }
+
+  @Override
+  public AvgPair getInitialAggregatedValue(@Nullable Object rawValue) {
+    if (rawValue == null) {
+      return new AvgPair();
+    }
+    if (rawValue instanceof byte[]) {
+      return deserializeAggregatedValue((byte[]) rawValue);
+    } else {
+      return processMultiValueArray(rawValue);
+    }
+  }
+
+  @Override
+  public AvgPair applyRawValue(AvgPair value, Object rawValue) {
+    if (rawValue instanceof byte[]) {
+      value.apply(deserializeAggregatedValue((byte[]) rawValue));
+    } else {
+      AvgPair mvResult = processMultiValueArray(rawValue);
+      value.apply(mvResult);
+    }
+    return value;
+  }
+
+  @Override
+  public AvgPair applyAggregatedValue(AvgPair value, AvgPair aggregatedValue) {
+    value.apply(aggregatedValue);
+    return value;
+  }
+
+  @Override
+  public AvgPair cloneAggregatedValue(AvgPair value) {
+    return new AvgPair(value.getSum(), value.getCount());
+  }
+
+  @Override
+  public boolean isAggregatedValueFixedSize() {
+    return true;
+  }
```

Comment:
This is used to determine whether a value aggregator can be used for ingestion aggregations - have you verified if these MV value aggregators work for ingestion aggregations too or only for star-tree indexes?

Reply:
I have now verified for realtime ingestion aggregation by quickly modifying `dailySales` table/schema config and raw data, it works fine. - https://github.com/kmozaid/pinot/commit/5710e79187d75a9f6552e39458cccf29c4364cbd